test/distributed/_tensor/test_tensor_ops.py

```diff
@@ -253,6 +253,17 @@ def test_equal(self):
         eq_result = dist_tensor_1.equal(dist_tensor_2)
         # equal op all reduces each shard's local result
         self.assertFalse(eq_result)
+        self.assertTrue(dist_tensor_1.is_same_size(dist_tensor_2))
+
+        # test if sharding are different
+        replica_spec = [Replicate()]
+        global_input = torch.ones(4 * self.world_size, 4)
+        dist_tensor_3 = DTensor.from_local(
+            global_input, device_mesh, replica_spec, run_check=False
+        )
+
+        self.assertTrue(dist_tensor_1.equal(dist_tensor_3))
```

Comment:
should we add tests for cases where values actually are different, and especially, for the corner case where the difference only occurs on a remote shard and we notice it by doing the replication?

Reply:
added.